src/pages/blog/components/post-card/PostCard.tsx

```diff
@@ -0,0 +1,143 @@
+import { Link } from "react-router-dom";
+
+import { Avatar, Box, IconButton, Typography } from "@mui/material";
+import { useMediaQuery } from "@mui/material";
+import { Icon } from "@mui/material";
+import { format } from "date-fns";
+
+import CommentIcon from "@/assets/icons/comment.svg";
+import EyeIcon from "@/assets/icons/eye.svg";
+import imagePlaceholder from "@/assets/icons/image-placeholder.svg?url";
+import ShareIcon from "@/assets/icons/three-dots-vertical.svg";
+
+import { routePaths } from "@/constants/routePaths";
+import useLazyImage from "@/hooks/use-lazy-mage/useLazyImage";
+import PostCardSkeleton from "@/pages/blog/components/post-card-skeleton/PostCardSkeleton";
+import {
+  BoxImageStyled,
+  BoxStyled,
+  ContentBox,
+  ImgStyled,
+  PostContentBox,
+  PostInfoBox,
+  StatsBox,
+} from "@/pages/blog/components/post-card/PostCard.styled";
+import LikeWidget from "@/pages/post/components/like-widget/LikeWidget";
+import theme from "@/theme/theme";
+
+type Post =
+  | {
+      id: number;
+      author: {
+        id: number;
+        username: string;
+        role: string;
+      };
+      createdAt: string;
+      updatedAt: string;
+      estimatedReadTime: number;
+      image: string;
+      title: string;
+      summary: string;
+      viewsCount: number;
+      commentsCount: number;
+    }
+  | undefined;
+
+// temporary post for ui
+const post: Post = {
+  id: 1,
+  author: {
+    id: 1,
+    username: "Admin",
+    role: "ADMIN",
+  },
+  createdAt: "2024-11-14T07:35:34.842Z",
+  updatedAt: "2024-11-14T07:35:34.842Z",
+  estimatedReadTime: 1,
+  image:
+    "https://static.wixstatic.com/media/84770f_170242c7269d4ba2ba8ad50591e1a1e8~mv2_d_4500_2992_s_4_2.jpg/v1/fill/w_925,h_615,al_c,q_85,usm_0.66_1.00_0.01,enc_auto/84770f_170242c7269d4ba2ba8ad50591e1a1e8~mv2_d_4500_2992_s_4_2.jpg",
+  title: "Why are Facials a Must for the Modern Woman",
+  summary:
+    "Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your audience to continue reading...",
+  viewsCount: 0,
+  commentsCount: 0,
+};
+// let post: Post = undefined;
+
+export default function PostCard() {
+  const isSmallScreen = useMediaQuery(theme.breakpoints.down("sm"));
+
+  if (post === undefined) return <PostCardSkeleton />;
+
+  const [image, { hasError }] = useLazyImage({
+    src: post.image,
+    placeholderSrc: imagePlaceholder,
+  });
+
+  return (
+    <BoxStyled>
+      <Link to={`${routePaths.posts.path}/${post.id}`}>
+        <BoxImageStyled isLoading={false} shouldShowImagePlaceholder={hasError}>
+          <ImgStyled src={image} alt={`'${post?.title}' post image`} />
+        </BoxImageStyled>
+        <ContentBox>
+          <PostInfoBox>
+            <Box display="flex" gap={1}>
+              <Avatar />
```

Comment:
Avatar should redirect to member by id page

Reply:
Fixed
Added link to author details

![image](https://github.com/user-attachments/assets/f168f7e7-84ba-494b-8919-aa537e3dfddf)